diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -0,0 +1,161 @@
+        }
+    }
+
+    private fun checkFileHasCode(node: ASTNode): Boolean {
+        val codeTokens = TokenSet.andNot(TokenSet.ANY, TokenSet.create(WHITE_SPACE, KDOC, BLOCK_COMMENT, EOL_COMMENT, PACKAGE_DIRECTIVE, IMPORT_LIST))
+        val hasCode = node.getChildren(codeTokens).isNotEmpty()
+        if (!hasCode) {
+            FILE_CONTAINS_ONLY_COMMENTS.warn(configRules, emitWarn, isFixMode, fileName, node.startOffset)
+        }
+        return hasCode
+    }
+
+    private fun checkCodeBlocksOrderAndEmptyLines(node: ASTNode) {
+        val copyrightComment = node.findChildBefore(PACKAGE_DIRECTIVE, BLOCK_COMMENT)
+        val headerKdoc = node.findChildBefore(PACKAGE_DIRECTIVE, KDOC)
+        val fileAnnotations = node.findChildByType(FILE_ANNOTATION_LIST)
+        // the following two nodes are always present, even if their content is empty
+        // also kotlin compiler itself enforces their position in the file
+        val packageDirectiveNode = node.findChildByType(PACKAGE_DIRECTIVE)!!
+        val importsList = node.findChildByType(IMPORT_LIST)!!
+
+        // checking order
+        listOfNotNull(copyrightComment, headerKdoc, fileAnnotations).forEach { astNode ->
+            val (afterThisNode, beforeThisNode) = astNode.getSiblingBlocks(copyrightComment, headerKdoc, fileAnnotations, packageDirectiveNode)
+            val isPositionIncorrect = (afterThisNode != null && !node.isChildAfterAnother(astNode, afterThisNode))
+                    || !node.isChildBeforeAnother(astNode, beforeThisNode)
+
+            if (isPositionIncorrect) {
+                FILE_INCORRECT_BLOCKS_ORDER.warnAndFix(configRules, emitWarn, isFixMode, astNode.text.lines().first(), astNode.startOffset) {
+                    node.moveChildBefore(astNode, beforeThisNode, 2)
+                }
+            }
+        }
+
+        // checking empty lines
+        arrayOf(copyrightComment, headerKdoc, fileAnnotations, packageDirectiveNode, importsList).forEach { astNode ->
+            astNode?.treeNext?.apply {
+                if (elementType == WHITE_SPACE && text.count { it == '\n' } != 2) {
+                    FILE_NO_BLANK_LINE_BETWEEN_BLOCKS.warnAndFix(configRules, emitWarn, isFixMode, astNode.text.lines().first(), astNode.startOffset) {
+                        (this as LeafPsiElement).replaceWithText("\n\n${text.replace("\n", "")}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun checkImportsOrder(node: ASTNode) {
+        val imports = node.getChildren(TokenSet.create(IMPORT_DIRECTIVE)).toList()
+
+        // importPath can be null if import name cannot be parsed, which should be a very rare case, therefore !! should be safe here
+        imports.filter { (it.psi as KtImportDirective).importPath!!.isAllUnder }.forEach {
+            FILE_COLLAPSED_IMPORTS.warn(configRules, emitWarn, isFixMode, it.text, it.startOffset)
+        }
+
+        val sortedImports = imports.sortedBy { it.text }
```

Comment:
I forgot - we decided simply to order them alphabetically?

Reply:
Yes, as a default sorting rule. When we write final version of more complicated sorting, it should be added as separate option.